src/main/java/uk/gov/justice/digital/service/DataStorageService.java

```diff
@@ -330,12 +308,14 @@ private void collectDeltaTablePaths(SparkSession spark, FileSystem fs, String ro
         logger.debug("Listing all delta table paths below: {}", rootPath);
         val path = new Path(rootPath);
 
-        Map<Boolean, List<String>> deltaTablesAndOtherDirs = Arrays.stream(fs.listStatus(path))
-                .filter(FileStatus::isDirectory)
-                .map(FileStatus::getPath)
-                .map(Path::toUri)
-                .map(URI::toString)
-                .collect(Collectors.partitioningBy(tablePath -> DeltaTable.isDeltaTable(spark, tablePath)));
+        Map<Boolean, List<String>> deltaTablesAndOtherDirs = (depthLimit == 0) ?
+                Collections.singletonMap(true, Collections.singletonList(rootPath)) :
+                Arrays.stream(fs.listStatus(path))
+                        .filter(FileStatus::isDirectory)
+                        .map(FileStatus::getPath)
+                        .map(Path::toUri)
+                        .map(URI::toString)
+                        .collect(Collectors.partitioningBy(tablePath -> DeltaTable.isDeltaTable(spark, tablePath)));
```

Comment:
When the depthLimit is 0 we use the provided rootPath as the delta table path. Otherwise, we recursively explore delta table paths.